streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -810,6 +813,36 @@ private StandbyTask findMatchingSuspendedStandbyTask(final TaskId taskId, final
         return null;
     }
 
+    private void closeNonAssignedSuspendedTasks() {
+        final Set<TaskId> currentActiveTaskIds = partitionAssignor.activeTasks().keySet();
+        final Iterator<Map.Entry<TaskId, StreamTask>> activeTaskIterator = suspendedActiveTasks.entrySet().iterator();
+        while (activeTaskIterator.hasNext()) {
+            closeAndRemoveIfNotAssigned(currentActiveTaskIds, activeTaskIterator.next(), activeTaskIterator);
```

Comment:
Could we return a boolean indicating if this task is removed or not? Then in line 964 below we can instead check that this map should then be empty since we have removed both recycled tasks or non-recycled ones (in line 863).